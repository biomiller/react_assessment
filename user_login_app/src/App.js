import React, { Component } from 'react';
import { BrowserRouter as Router, Route } from "react-router-dom";
import axios from "axios";

import {
  Col,
  Row
} from 'reactstrap';

import { TopNavbar } from "./components/TopNavbar.js";
import { UserList } from "./components/UserList.js"
import { Register } from "./components/Register.js"
import { Login } from "./components/Login.js"




export class App extends Component {

  constructor() {
    super();
    this.state = {
      loggedIn: "Hello Guest",
      data: []
    }
  }

  getData = () => {
    axios
      .get("http://localhost:5000/user/all")
      .then(response => {
        this.setState({
          data: response.data.reverse()
        });
      });
  };

  userLoggedIn = (username) => {
    this.setState({
      loggedIn: "Hello " + username
    })
  }

  componentDidMount = () => {
    this.getData()
  }

  render() {
    return (
      <div>
        <Router>
          <Row>
          <TopNavbar loggedIn={this.state.loggedIn} />
          </Row>

          <Row>
          <Col sm="12" md={{ size: 6, offset: 6 }}>
            <Route exact path={`/`} render={() => <h1>Home Page</h1>} />
            <Route exact path={`/ViewUsers`} render={() => <UserList data={this.state.data} />} />
            <Route exact path={`/Register`} render={() => <Register getData={this.getData} userLoggedIn={this.userLoggedIn} />} />
            <Route exact path={`/Login`} render={() => <Login getData={this.getData} userLoggedIn={this.userLoggedIn} />} />
          </Col>
          </Row>


        </Router>
      </div>
    );
  }
}

export default App;
