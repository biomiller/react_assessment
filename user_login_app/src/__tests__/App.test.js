import React from 'react';
import renderer from 'react-test-renderer';

import { MemoryRouter } from 'react-router-dom';

import { App } from '../App.js';

it('is app rendered', () => {
  const component = renderer
  .create(
    <MemoryRouter>
      <App />
    </MemoryRouter>
  )
  .toJSON();

  expect(component).toMatchSnapshot();
})
