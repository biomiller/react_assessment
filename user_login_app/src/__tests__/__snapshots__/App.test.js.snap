// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`is app rendered 1`] = `
<div>
  <div
    className="row"
  >
    <div>
      <nav
        className="navbar navbar-expand-md navbar-light bg-light"
      >
        <a
          href="/"
          onClick={[Function]}
        >
          <a
            className="navbar-brand"
          >
            React Assessment
          </a>
        </a>
        <a
          className="navbar-brand"
        >
          Hello Guest
        </a>
        <ul
          className="nav"
        >
          <li
            className="nav-item"
          >
            <a
              href="/Register"
              onClick={[Function]}
            >
              <button
                aria-label={null}
                className="btn btn-secondary"
                onClick={[Function]}
              >
                Register
              </button>
            </a>
            <a
              href="/Login"
              onClick={[Function]}
            >
              <button
                aria-label={null}
                className="btn btn-secondary"
                onClick={[Function]}
              >
                Login
              </button>
            </a>
            <a
              href="/ViewUsers"
              onClick={[Function]}
            >
              <button
                aria-label={null}
                className="btn btn-secondary"
                onClick={[Function]}
              >
                View Users
              </button>
            </a>
          </li>
        </ul>
      </nav>
    </div>
  </div>
  <div
    className="row"
  >
    <div
      className="col-sm-12 col-md-6 offset-md-6"
    >
      <h1>
        Home Page
      </h1>
    </div>
  </div>
</div>
`;
